### DISCUSSION TITLE
How to add requires in script mode?

### DISCUSSION THREAD

**Discussion by shaqtsui** [xmake-io/xmake#3483] [Q&A]
I need to install a package base on gcc version. 
Detection api is not avialable in description scope,  add require api is not available in scrip scope. (embarass face)

Try ask bing, got (for fun):
![image](https://user-images.githubusercontent.com/4743082/223931016-680d97dd-efce-448b-b750-4ef6fb720346.png)

**Comment by waruqi** [comment]
use custom option + on_check to detect gcc version, then `if has_config("optname") then add_requires` in description scope.

**Reply by shaqtsui** [reply]
Nice trick :D
